index.html

```diff
@@ -232,39 +235,393 @@ <h3>Create</h3>
+<em>signature verification method</em>.
+            </li>
+            <li>
+If <em>options</em> `disableKeyAgreement` is not set to `true`,
+add the <em>encryption verification method</em> value to the
+`verificationMethod` array and initialize the `keyAgreement` property in
+<em>document</em> to an array where the first item is the value of the `id`
+property in <em>encryption verification method</em>.
+            </li>
+            <li>
+Return <em>document</em>.
+            </li>
+          </ol>
+          <p>
+Any alternative algorithm can be used that produces an equivalent result to the
+output from the algorithm above.
+          </p>
+        </section>
+
+        <section class="normative">
+          <h4>Signature Method Creation Algorithm</h4>
+
+          <p>
+The following algorithm can be used for decoding a multibase-encoded multicodec
+value into a verification method that is suitable for verifying digital
+signatures. An <em>identifier</em>, <em>multibase value</em>, and
+<em>options</em> are the required inputs. A <em>verification method</em> is
+the resulting output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Initialize <em>verification method</em> to an empty object.
+            </li>
+            <li>
+Set <em>multicodec value</em> and <em>raw public key bytes</em> to the result
+of passing <em>multibase value</em> and <em>options</em> to
+<a href="#decode-public-key-algorithm"></a>.
+            </li>
+            <li>
+Ensure the proper key length of <em>raw public key bytes</em> based on the
+<em>multicodec value</em> table provided below:
+              <table class="simple">
+                <thead>
+                  <th>Multicodec hexadecimal value</th>
+                  <th>public key byte length</th>
+                  <th>Description</th>
+                </thead>
+
+                <tbody>
+                  <tr>
+                    <td>`0xe7`</td>
+                    <td>33 bytes</td>
+                    <td>secp256k1-pub - Secp256k1 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0xec`</td>
+                    <td>32 bytes</td>
+                    <td>x25519-pub - Curve25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0xed`</td>
+                    <td>32 bytes</td>
+                    <td>ed25519-pub - Ed25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1200`</td>
+                    <td>33 bytes</td>
+                    <td>p256-pub - P-256 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1201`</td>
+                    <td>?? bytes</td>
+                    <td>p384-pub - P-384 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1202`</td>
+                    <td>?? bytes</td>
```

Comment:
```suggestion
                    <td> 65 bytes</td>
```

https://stackoverflow.com/questions/50002149/why-p-521-public-key-x-y-some-time-is-65-bytes-some-time-is-66-bytes#:~:text=521%20bits%20is%2065%20full,byte%20is%200x00%20or%200x01).

Reply:
seems wrong tho... pretty sure that for P-256.

Reply:
Yeah, I found that article too, but couldn't tell if they were talking about a compressed value (most likely) or some internal Go representation (which would surprise me if it was a compressed value). In any case, probably have to use the openssl trick [mentioned here](https://github.com/w3c-ccg/did-method-key/pull/51/files#r895715511).

Reply:
cc @clehner @troyronda halp!